tests/python_package_test/test_basic.py

```diff
@@ -609,3 +611,57 @@ def test_custom_objective_safety():
     good_bst_multi.update(fobj=_good_gradients)
     with pytest.raises(ValueError, match=re.escape(f"number of models per one iteration ({nclass})")):
         bad_bst_multi.update(fobj=_bad_gradients)
+
+@pytest.fixture(name="fake_model")
+def _fake_model() -> lgb.Booster:
+    # TODO: maybe removed deps on data_dir later
+    data_dir = Path(__file__).parent.parent.parent / "examples/binary_classification"
+
+    df_train = pd.read_csv(data_dir / "binary.train", header=None, sep="\t")
+    weights = pd.read_csv(data_dir / "binary.train.weight", header=None)[0]
+
+    X_train = df_train.drop(0, axis=1)
+    y_train = df_train[0]
+
+    params = {
+        "boosting_type": "gbdt",
+        "objective": "binary",
+        "metric": "binary_logloss",
+        "num_leaves": 31,
+        "learning_rate": 0.05,
+        "feature_fraction": 0.9,
+        "bagging_fraction": 0.8,
+        "bagging_freq": 5,
+        "verbose": 0,
+    }
+    lgb_train = lgb.Dataset(X_train, y_train, weight=weights, free_raw_data=False)
+    feature_name = [f"feature_{col}" for col in X_train.columns]
+
+    gbm = lgb.train(
+        params,
+        lgb_train,
+        num_boost_round=3,
+        valid_sets=lgb_train,  # eval training data
+        feature_name=feature_name,
+        categorical_feature=[21],
+    )
+
+    return gbm
+
+def test_booster_load_params_when_passed_model_file(fake_model: lgb.Booster) -> None:
+    with tempfile.TemporaryDirectory() as temp_dir:
+        model_file = Path(temp_dir) / "model.txt"
+        fake_model.save_model(model_file)
+
+        loaded = lgb.Booster(model_file=model_file)
+
+    # TODO: needs parse more params
+    assert loaded.params['boosting'] == 'gbdt'
```

Comment:
As you continue working on this feature, please consider the following additional parameters as a minimal sample that should help catch some different types of parsing issues.

* `monotone_constraints`: an array of ints
    - https://github.com/microsoft/LightGBM/blob/a06fadfb7ac3fdb26da3d4afc061a8b976070c50/src/io/config_auto.cpp#L448
* `learning_rate`: a double
    - https://github.com/microsoft/LightGBM/blob/a06fadfb7ac3fdb26da3d4afc061a8b976070c50/src/io/config_auto.cpp#L332
* `boost_from_average`: a boolean
    - https://github.com/microsoft/LightGBM/blob/a06fadfb7ac3fdb26da3d4afc061a8b976070c50/src/io/config_auto.cpp#L583